File: app/src/main/java/com/sortedqueue/learnandroid/asynctasks/CodeFileReaderTask.java
package com.sortedqueue.learnandroid.asynctasks;

import android.content.Context;
import android.os.AsyncTask;
import android.view.View;
import android.widget.ProgressBar;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.nio.charset.Charset;

/**
 * Created by Alok on 02/06/17.
 */

public class CodeFileReaderTask extends AsyncTask<Void, Void, Void> {

    private Context context;
    private String code;
    private String fileId;

    public interface OnDataReadListener {
        void onDataReadComplete( String code );
    }

    //Interface to communicate back the response to UI
    private OnDataReadListener onDataReadListener;
    private ProgressBar progressBar;

    public CodeFileReaderTask(ProgressBar codeProgressBar, Context context, String fileId, OnDataReadListener onDataReadListener) {
        this.context = context;
        this.fileId = fileId;
        this.onDataReadListener = onDataReadListener;
        this.progressBar = codeProgressBar;
    }

    @Override
    protected Void doInBackground(Void... voids) {
        readFileAsList(context);
        return null;
    }

    @Override
    protected void onPreExecute() {
        super.onPreExecute();
        if( progressBar != null ) {
            progressBar.setVisibility(View.VISIBLE);
        }
    }

    @Override
    protected void onPostExecute(Void aVoid) {
        super.onPostExecute(aVoid);
        if( progressBar != null ) {
            progressBar.setVisibility(View.GONE);
        }
        if( onDataReadListener != null ) {
            onDataReadListener.onDataReadComplete(code);
        }
    }

    private void readFileAsList(Context context) {
        String line;
        try {

            InputStream inputStream = context.getResources().openRawResource(
                    context.getResources().getIdentifier(fileId,
                            "raw", context.getPackageName()));
            InputStreamReader inputStreamReader = new InputStreamReader(inputStream, Charset.forName("UTF-8"));
            BufferedReader bufferedReader = new BufferedReader(inputStreamReader);
            code = "";
            while ((line = bufferedReader.readLine()) != null) {
                code += line + "\n";
            }

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/sortedqueue/learnandroid/asynctasks/CodeFileReaderTask.kt
package com.sortedqueue.learnandroid.asynctasks

import android.annotation.SuppressLint
import android.content.Context
import android.os.AsyncTask
import android.view.View
import android.widget.ProgressBar

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader
import java.nio.charset.Charset

/**
 * Created by Alok on 02/06/17.
 */
@SuppressLint("StaticFieldLeak")
class CodeFileReaderTask( private val progressBar: ProgressBar?, private val context: Context, private val fileId: String, //Interface to communicate back the response to UI
                         private val onDataReadListener: OnDataReadListener?) : AsyncTask<Void, Void, Void?>() {
    private var code: String? = null

    interface OnDataReadListener {
        fun onDataReadComplete(code: String?)
    }

    override fun doInBackground(vararg voids: Void): Void? {
        readFileAsList(context)
        return null
    }

    override fun onPreExecute() {
        super.onPreExecute()
        if (progressBar != null) {
            progressBar.visibility = View.VISIBLE
        }
    }

    override fun onPostExecute(aVoid: Void?) {
        super.onPostExecute(aVoid)
        if (progressBar != null) {
            progressBar.visibility = View.GONE
        }
        onDataReadListener?.onDataReadComplete(code)
    }

    private fun readFileAsList(context: Context) {
        var line: String?
        try {

            val inputStream = context.resources.openRawResource(
                    context.resources.getIdentifier(fileId,
                            "raw", context.packageName))
            val inputStreamReader = InputStreamReader(inputStream, Charset.forName("UTF-8"))
            val bufferedReader = BufferedReader(inputStreamReader)
            code = ""
            line = bufferedReader.readLine()
            while ((line) != null) {
                code += line + "\n"
                line = bufferedReader.readLine()
            }

        } catch (e: IOException) {
            e.printStackTrace()
        }

    }
}
